proposals/0000-proposal-process.rst

```diff
@@ -0,0 +1,319 @@
+already exists and a significant fraction of GHC developers are already quite
+comfortable with it. The fact that we already use it for Wiki and ticket
+tracking means that cross-referencing between proposals, tickets, and the Wiki
+is trivial. Any replacement system would almost certainly regress in this area.
+
+Alternatives
+------------
+
+There are a few alternatives which are worth considering,
+
+* We continue to build on Trac, but attempt to be more rigorous with our
+  current protocol. Namely we attempt to better document and more consistently
+  enforce the procedures we currently have.
+
+* Adopting the process discussed above but replacing GitHub with Phabricator.
+  This would offer the disadvantage of being slightly harder for casual
+  contributors. Moreover, Phabricator lacks some of GitHub's support for
+  formatting markup. Phabricator, however, offers the advantage of integrating
+  better with GHC's current Trac instance and avoid adding yet another tool to
+  the GHC developer's toolchain.
+
+* Adopting the process discussed above but replacing GitHub with GitLab. GitLab
+  has many of the features of GitHub but with the advantage of being open-source
+  and self-hostable. Unfortunately it would mean introducing yet another tool
+  into the GHC developer's toolbelt and add yet another piece of infrastructure
+  to administer.
+
+* We move to something closer to the Python PEP scheme. Here a committee is
+  formed for each proposal; the committee is responsible for crafting and
+  collecting feedback on the proposal. The final acceptance decision is made by
+  the Benevolant Dictator for Life. Discussion can be carried out on a medium
+  of the committee's choosing, with a common choice being a specially-created
+  mailing list.
+
+* The Rust community has been `considering
+  <https://aturon.github.io/blog/2016/07/05/rfc-refinement/>`_ breaking their
+  process up in multiple stages to make their proposal queue more manageable. To
+  first order these stages are,
+
+  1. form a consensus around the problem and the need to solve it
+  2. propose and refine possible solutions
+  3. choose from among the proposed solutions
+
+  The separation of step (1) from the others gives the community a means of
+  concentrating focus on a set of small problems and encouraging exploration
+  down multiple solution avenues. The cost of this is that it may frustrate
+  authors with a slower, more involved process.
+
+* Something else entirely...
+
+
+Moving closer to the Rust process
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Also, there are a few facets of the Rust process which the proposed process does
+not carry over for a variety of reasons:
+
+* *Shepherds*. In the Rust process each submitted proposal is assigned a
+  shepherd. This is a trusted core developer who is charged with keeping the
+  proposal moving through the process. At the moment GHC arguably lacks the
+  contributor pool to guarantee this.
```

Comment:
What's a concrete difference between what this says and what the right ghc
contributors / experts would be?

Either way, I agree with Gershom, let's just try stuff and see what
happens.  Anything else isn't moving the needle ;)

On Wednesday, July 20, 2016, Adam C. Foltzer <notifications@github.com>
wrote:

> In proposals/0000-proposal-process.rst
> <https://github.com/ghc-proposals/ghc-proposals/pull/1#discussion_r71606567>
> :
>
> > +  down multiple solution avenues. The cost of this is that it may frustrate
> > +  authors with a slower, more involved process.
> > +
> > +* Something else entirely...
> > +
> > +
> > +Moving closer to the Rust process
> > +~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> > +
> > +Also, there are a few facets of the Rust process which the proposed process does
> > +not carry over for a variety of reasons:
> > +
> > +* *Shepherds*. In the Rust process each submitted proposal is assigned a
> > +  shepherd. This is a trusted core developer who is charged with keeping the
> > +  proposal moving through the process. At the moment GHC arguably lacks the
> > +  contributor pool to guarantee this.
>
> Is it necessary that a shepherd be a core developer? We certainly wouldn't
> want strangers in this role, but I don't think we should limit this to
> people with hands-on GHC hacking experience. It seems far more important to
> have an understanding of community norms and processes, and to know whom to
> consult when more technical depth is required.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ghc-proposals/ghc-proposals/pull/1/files/1c57c641f19b3f4ca00f6525dd37a44b5042f702#r71606567>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAQwqViHdeXxKEGxvXbEmRqCKHm9hSXks5qXo2RgaJpZM4JIsl->
> .
>